src/test/resources/itemsTests.xml

```diff
@@ -0,0 +1,16 @@
+<suite name="Suite2" verbose="1">
+    <test name="ItemsTests">
+        <packages>
+            <package name="com.coherent.testng"/>
+        </packages>
+        <groups>
+            <run>
+                <include name="Regression.Cart"/>
+                <exclude name="Regression.VirtualItem"/>
+            </run>
+        </groups>
+    </test>
+</suite>
+
+<!--Question:-->
+<!--Here JsonParserTNG test fails. However it doesn't fail from JsonParserTNG class. I'm wondering why-->
```

Comment:
When you include specific group name (like Regression.Cart) it means that testNG will execute only tests that contains that group. If before/after doesn't has such group it will not be executed. 
Test that fails by some reason validated total price against number 72. According to you code it looks like that was executed the third in the list so 3 times realItem was added to cart. Due to reason that when you delete realItem the price doesn't change, you test was green when you executed it in class. But it fails there due to reason that test executed first and without before/after methods